src/components/ChatView.vue

```diff
@@ -19,15 +19,38 @@
   -->
 
 <template>
-	<div class="chatView">
-		<MessagesList :token="token" />
-		<NewMessageForm />
+	<div
+		class="chatView"
+		@dragover.prevent="isDraggingOver = true"
+		@dragleave.prevent="isDraggingOver = false"
+		@drop.prevent="processFiles">
```

Comment:
Hover state shows for guests and triggers a basic auth login mask by the browser afterwards.
Either we show "You need to be logged in to upload files" in the drop zone, or the drop zone should not show up at all.